src/app/pages/bitcoin-contract-list/bitcoin-contract-list.tsx

```diff
@@ -0,0 +1,53 @@
+import { useState } from 'react';
+
+import { Flex, Text } from '@stacks/ui';
+
+import {
+  BitcoinContractListItem,
+  useBitcoinContracts,
+} from '@app/common/hooks/use-bitcoin-contracts';
+import { useOnMount } from '@app/common/hooks/use-on-mount';
+import { FullPageLoadingSpinner } from '@app/components/loading-spinner';
+
+import { BitcoinContractListItemLayout } from './components/bitcoin-contract-list-item-layout';
+import { BitcoinContractListLayout } from './components/bitcoin-contract-list-layout';
+
+export function BitcoinContractList() {
+  const { getAllSignedBitcoinContracts, formatBitcoinContracts } = useBitcoinContracts();
+  const [bitcoinContracts, setBitcoinContracts] = useState<BitcoinContractListItem[]>([]);
+  const [isLoading, setLoading] = useState(true);
+
+  useOnMount(() => {
+    const fetchAndFormatBitcoinContracts = async () => {
+      const fetchedBitcoinContracts = await getAllSignedBitcoinContracts();
+      const formattedBitcoinContracts = formatBitcoinContracts(fetchedBitcoinContracts);
+      setBitcoinContracts(formattedBitcoinContracts);
+      setLoading(false);
+    };
+    fetchAndFormatBitcoinContracts();
+  });
+
+  if (isLoading) return <FullPageLoadingSpinner />;
+
+  return (
+    <BitcoinContractListLayout>
+      {bitcoinContracts.length === 0 ? (
+        <Flex width="100%" height="100vh" alignItems="center" justifyContent="center">
+          <Text textAlign="center">You don't have any open Bitcoin Contracts.</Text>
+        </Flex>
+      ) : (
+        bitcoinContracts.map(bitcoinContract => {
+          return (
+            <BitcoinContractListItemLayout
+              key={bitcoinContract.id}
+              id={bitcoinContract.id}
```

Comment:
Re-earlier comment

```suggestion
              id={truncateMiddle(bitcoinContract.id)}
```

this would make more sense to me as I wouldn't expect the data to be pre-formatted

Reply:
Removed the format function, thank you for suggesting!